src/js/node/url.js

```diff
@@ -315,7 +316,7 @@ Url.prototype.parse = function (url, parseQueryString, slashesDenoteHost) {
        * have non-ASCII characters, i.e. it doesn't matter if
        * you call it with a domain that already is ASCII-only.
        */
-      this.hostname = new URL("http://" + this.hostname).hostname;
+      this.hostname = toASCII(this.hostname);
```

Comment:
Reference
https://github.com/nodejs/node/blob/c25878d370c6d96abb382d9ee10d1838da4ec29d/lib/url.js#L400-L404

Reply:
can this be lazily loaded? something like

```ts
// at the top of the file:
let toASCII;
```

```suggestion
      this.hostname = (toASCII ??= require("node:punycode").toASCII)(this.hostname);
```

Reply:
Done! Thanks for reviewing.